[IMP] website, website_sale, *: add links to documentation in the settings

jcs-odoo opened the pull request:
This commit adds buttons to existing documentation pages next to the features' labels in the settings.

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/160135).

AntoineVDV commented on addons/website_sale/views/res_config_settings_views.xml:
```diff
@@ -235,7 +235,8 @@
 
             <setting id="google_analytics_setting" position="after">
-                <setting id="autocomplete_googleplaces_setting" help="Use Google Places API to validate addresses entered by your visitors">
+                <setting id="autocomplete_googleplaces_setting" help="Use Google Places API to validate addresses entered by your visitors"
+                    documentation="/applications/websites/website/configuration/address_autocomplete.html">
```
Easier to read.
```suggestion
                <setting
                    id="autocomplete_googleplaces_setting"
                    help="Use Google Places API to blablabla
                    documentation="/applications/websites/etc/coincoin.html"
                >
```

jcs-odoo commented on addons/website/views/res_config_settings_views.xml:
```diff
@@ -103,15 +112,14 @@
                                         required="has_google_analytics"/>
                                 </div>
                             </div>
-                            <div invisible="not has_google_analytics">
-                                <a href="https://www.odoo.com/documentation/17.0/applications/websites/website/optimize/google_analytics.html"
-                                        class="oe_link" target="_blank">
-                                    <i class="oi oi-arrow-right"/>
-                                    How to get my Measurement ID
-                                </a>
-                            </div>
```
As discussed with the PO

AntoineVDV commented:
The website team will review their scope, but for eCommerce: @robodoo delegate+ just in case

jcs-odoo commented on addons/website/views/res_config_settings_views.xml:
```diff
@@ -29,6 +33,7 @@
                             <div class="content-group">
                                 <div class="row mt8">
                                     <label class="col-lg-3" string="Domain" for="website_domain"/>
+                                    <a href="https://www.odoo.com/documentation/17.0/applications/websites/website/configuration/domain_names.html#domain-name-website-map" title="Documentation" class="o_doc_link" target="_blank"></a>
```
Auto-review:
It is not well displayed
<img width="583" alt="image" src="https://github.com/odoo/odoo/assets/46810843/97096e3f-87ff-491e-bf43-007ea0838d0f">

rdeodoo commented on addons/website/views/res_config_settings_views.xml:
```diff
@@ -174,13 +190,6 @@
                                         required="has_plausible_shared_key"/>
                                 </div>
                             </div>
-                            <div invisible="not has_plausible_shared_key">
```
:point_up:

rdeodoo commented on addons/website/views/res_config_settings_views.xml:
```diff
@@ -103,15 +112,14 @@
                                         required="has_google_analytics"/>
                                 </div>
                             </div>
-                            <div invisible="not has_google_analytics">
```
:point_up:

rdeodoo reviewed (changes requested):
Hello, we can't remove nodes in stable, it will need to be re-targeted to master.
Can you ping us once it's done and once you have fixed your "auto-review" point?

I'll set back the PR to draft (it's not needed anymore but an old habit just in case..) so you can change the PR target and push force your branch after having it "rebased" on master :+1: 
(Don't create a new PR)

jcs-odoo commented:
> Hello, we can't remove nodes in stable, it will need to be re-targeted to master. Can you ping us once it's done and once you have fixed your "auto-review" point?
> 
> I'll set back the PR to draft (it's not needed anymore but an old habit just in case..) so you can change the PR target and push force your branch after having it "rebased" on master 👍 (Don't create a new PR)

Is it okay to leave this PR in 17.0 and remove the nodes only in the master fw-port?

jcs-odoo commented on addons/website/views/res_config_settings_views.xml:
```diff
@@ -20,23 +20,33 @@
             <xpath expr="//form" position="inside">
                 <app data-string="Website" string="Website" name="website" groups="website.group_website_designer">
-                    <setting type="header" string="Settings of Website">
+                    <setting
+                        type="header"
+                        string="Settings of Website"
+                        documentation="/applications/websites/website/configuration/multi_website.html"
+                    >
```
@rdeodoo the doc link doesn't show up, is it because this setting has a specific class? How should I fix it? (or maybe I shouldn't and just remove the doc link ^^' )

rdeodoo replied:
I guess, this `type="header"` is a very specific thing, used only in website and POS and a third one I believe.
It's probably not correctly supporting this kind of doc (as other setting blocks not converted by the framework recent refactoring)

jcs-odoo commented on addons/website/views/res_config_settings_views.xml:
```diff
@@ -20,23 +20,33 @@
                                 </div>
                                 <div class="row mt8">
-                                    <label class="col-lg-3" string="Languages" for="language_ids"/>
+                                    <div class="col-lg-3">
+                                        <label string="Languages" for="language_ids"/>
+                                        <a href="https://www.odoo.com/documentation/17.0/applications/websites/website/configuration/translate.html" title="Documentation" class="o_doc_link" target="_blank"></a>
+                                    </div>
```
@rdeodoo Since the div for the text is so small, the ? button goes to the next line. I checked with other languages and other settings and it's Odoo's default behavior, it's just that it doesn't look good in this specific case.

![image](https://github.com/odoo/odoo/assets/46810843/102c22c9-c976-46cd-b557-6ba29048e0b5)

rdeodoo commented on addons/website/views/res_config_settings_views.xml:
```diff
@@ -20,23 +20,33 @@
                             <div class="content-group">
                                 <div class="row mt8">
-                                    <label class="col-lg-3" string="Domain" for="website_domain"/>
+                                    <div class="col-lg-3">
```
we can't do that in stable, don't change DOM structure it will break xpath
honestly, just retarget to master, and probably then need for the framework to correctly handle those old settings, as discussed here https://discord.com/channels/678381219515465750/687338039717920792/1218166440314540082

rdeodoo commented on addons/website/views/res_config_settings_views.xml:
```diff
@@ -20,23 +20,33 @@
                                 </div>
                                 <div class="row mt8">
-                                    <label class="col-lg-3" string="Languages" for="language_ids"/>
+                                    <div class="col-lg-3">
```
same here

rdeodoo reviewed (changes requested):
Can you remove the structure change (and the doc at those places I guess)?
Then either target master only or keep here only the other changes.

jcs-odoo commented:
@rdeodoo I removed the doc links that were an issue here (it will be done separately) and changed the target to master nonetheless, because of the new doc links that are also modified (analytics links)

qsm-odoo reviewed (approved):
@robodoo r+
